src/shader.cpp

```diff
@@ -135,14 +143,17 @@ bool Shader::compileSourceCode(const QByteArray& src)
 
 //------------------------------------------------------------------------------
 // ShaderProgram implementation
-ShaderProgram::ShaderProgram(const QGLContext * context, QObject* parent)
+ShaderProgram::ShaderProgram(QObject* parent)
     : QObject(parent),
-    m_context(context),
     m_pointSize(5),
     m_exposure(1),
     m_contrast(1),
     m_selector(0)
-{ }
+{
+    m_vertexShader = NULL;
+    m_fragmentShader = NULL;
+    m_shaderProgram = NULL;
```

Comment:
These should be unnecessary (actually I'm surprised they even work) - `unique_ptr` automatically initializes the held pointer to null.